### PR TITLE
send test telemetry to unchecked method

### LINKED ISSUE #9218: Enable queries on test run history
I'd like to be able to easily check the run history of a test or general trends in test runs/pass rate.

Some cases I can think of:
- I can easily check if a "flaky test" issue is still relevant. There are plenty of old ones that could probably just be closed.
- If a test fails in CI, I can go check if that test has already been failing before I made my change.
- When troubleshooting a flaky test, I can get a better idea of when it started failing to track down relevant changes

Notes:
- The CI system doesn't send any telemetry, so we currently can't just emit our normal telemetry events
- The "publish test results" step in the github action sounds promising, but is currently failing for auth reasons

### PR DESCRIPTION
**Pull request by amunger**
Fixes #9218

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/telemetry/index.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export function _resetSharedProperties(): void {
 
 let telemetryReporter: TelemetryReporter | undefined;
 function getTelemetryReporter() {
-    if (!isTestExecution() && telemetryReporter) {
```
Comment (amunger) [inline src/telemetry/index.ts:92]: this doesn't change anything, but the check didn't make any sense to me - Create a new reporter each time if we're testing.

File: src/test/testHooks.ts [hunk 2]
```diff
@@ -1,20 +1,50 @@
...
+                Telemetry.RunTest,
+                {
+                    testName: this.currentTest?.title,
+                    testResult: result
+                },
+                measures
+            );
         }
+    },
+    AfterAll: async () => {
```
Comment (DonJayamanne) [inline src/test/testHooks.ts:42]: feels weird to have everythign camel cased, and this is pascalCased. I'd doubel check if its `afterAll` instead.

Comment (DonJayamanne) [reply src/test/testHooks.ts:42]: Confirmed, its meant to be `afterAll`.

File: src/test/testHooks.ts [hunk 3]
```diff
@@ -1,20 +1,50 @@
...
-import { sendTelemetryEvent } from '../telemetry';
+import { AppinsightsKey, JVSC_EXTENSION_ID, Telemetry } from '../platform/common/constants';
+import TelemetryReporter from '@vscode/extension-telemetry/lib/telemetryReporter';
 import { IS_CI_SERVER } from './ciConstants.node';
+import { extensions } from 'vscode';
+import { sleep } from '../platform/common/utils/async';
+
+let telemetryReporter: TelemetryReporter;
 
 export const rootHooks = {
```
Comment (DonJayamanne) [inline src/test/testHooks.ts:10]: types to the rescue..

```suggestion
export const rootHooks: : Mocha.RootHookObject = {
```

Comment (DonJayamanne) [reply src/test/testHooks.ts:10]: This should pick up other typos in the future.

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**